nodepool/templates/nodepool.yml.j2

```diff
@@ -159,6 +172,22 @@ providers:
             flavor-name: releng-7
             key-name: jenkins
             console-log: true
+      - name: onmetal
+        max-servers: 10
```

Comment:
what exactly does `max-servers` mean for a pool?

i mean, i presume that there will never be any more than 10 `ironic-storage-perf` nodes used from phobos at any time, but what is nodepool's allocation algorithm:
* allocate all 10 right away?
* allocate one at a time as needed, up to 10, and maintaining all 10 forevermore?
* allocate one at a time as needed, up to 10, but then releasing nodes after a period of lack of demand?

i ask because we only have 13 storage-perf nodes in phobos, with 9 of those currently in use, and i want to be prepared to explain when there are none available why jenkins is holding 10 of them but potentially not using them (eg "it's only been an hour or two since all the nightly jobs finished so the nodepool release timeout hasn't occurred, but the majority should be freed up in another hour or two.")

Reply:
@coreywright It means that nodepool will never use more than 10. It only allocates nodes when they are asked for by a test, or to satisfy the `min-ready` for the node type - but the `min-ready` can be satisfied by any of the regions.